packages/edit-site/src/components/secondary-sidebar/list-view-sidebar.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalBlockNavigationTree as BlockNavigationTree,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { closeSmall } from '@wordpress/icons';
+import { useShortcut } from '@wordpress/keyboard-shortcuts';
+import { focus } from '@wordpress/dom';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+
+/**
+ * Hook used to focus on mount the selected block's corresponding list item,
+ * or the first list item if no block is selected.
+ *
+ * @see packages/compose/src/hooks/use-focus-on-mount/index.js
+ *
+ * @return {Function} Ref callback.
+ */
+function useFocusSelectedOnMount() {
+	return useCallback( ( node ) => {
+		if ( ! node || node.contains( node.ownerDocument.activeElement ) ) {
+			return;
+		}
+
+		const selectedItem = node.querySelector(
+			'.block-editor-block-navigation-leaf.is-selected'
```

Comment:
instead of trying to find the element using classnames, can't we use a list of refs instead, since it seems it's all part of the same list view component?

Reply:
Actually... I think there's an even better way!

https://github.com/WordPress/gutenberg/blob/b3127331294c094ff115d1bf9b9d08439471b5d4/packages/block-editor/src/components/block-navigation/block.js#L82-L86

The code is already there, but it's coupled with the BlockNavigation's experimental features (e.g. drag and drop, a more menu for all the list items).
It just makes sense to reuse that for our purposes as well.